lib/serial_shape.ml

```diff
@@ -0,0 +1,257 @@
+open Import
+
+module Conv = struct
+  type ('repr, 'true_type) t = {
+    from_repr : 'repr -> ('true_type, Rresult.R.msg) result;
+    to_repr : 'true_type -> 'repr;
+    equal : ('true_type -> 'true_type -> bool) option;
+    pp : 'true_type Fmt.t option;
+  }
+
+  let make ~from_repr ~to_repr ?equal ?pp () = { from_repr; to_repr; equal; pp }
+end
+
+type _ t =
+  | Sbool : bool t
+  | Sstring : string t
+  | Slist : 'a t -> 'a list t
+  | Spair : 'a t * 'b t -> ('a * 'b) t
```

Comment:
It is to avoid conflict with `OpamParserTypes.value_kind` constructors which are meant to be converted using those.

Reply:
The code would be easier bloated with `OpamParserTypes.FullPos.` prefix to constructors or would be hard to rid since you would have two constructors with the same name used everywhere for different things.